### ISSUE TITLE
Learning rate goes lower than specified min_lr

### ISSUE THREAD

**Issue by lennarteing** [katsura-jp/pytorch-cosine-annealing-with-warmup#16]
I found that I can make the lr go lower than the specified min_lr for small learning rates. Even though at learning rates that small it doesn't make a huge difference, it is concerning me.

Code to reproduce:
```python
model = ...
optimizer = optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-5) # lr is min lr
scheduler = CosineAnnealingWarmupRestarts(optimizer,
                                          first_cycle_steps=30000,
                                          cycle_mult=1.0,
                                          max_lr=5e-4,
                                          min_lr=5e-6,
                                          warmup_steps=6000,
                                          gamma=0.5)
for iteration in range(300000):
    scheduler.step()
```

This will output a learning rate looking like this:
![image](https://github.com/katsura-jp/pytorch-cosine-annealing-with-warmup/assets/17054192/0acb3024-9830-48c1-bc0d-db9132a04d48)

Maybe I am wrong in my understanding wether or not learning rate should be changed within an epoch or not, however I expect the lr not to fall below 5e-6.
